diff --git a/CMakeLists.txt b/CMakeLists.txt
index 764ea962ee..6f401f72e7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,6 +53,11 @@ if(NOT CMAKE_CXX_EXTENSIONS)
 endif()
 
 # ====
+
+set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "/usr/bin/time -v -o ${TIME_LOG_FILE} -a")
+set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK "/usr/bin/time -v -o ${TIME_LOG_FILE} -a")
+set_property(GLOBAL PROPERTY RULE_LAUNCH_CUSTOM "/usr/bin/time -v -o ${TIME_LOG_FILE} -a")
+
 include(${CMAKE_CURRENT_SOURCE_DIR}/CMake/ITKInitializeBuildType.cmake)
 
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/CMake ${CMAKE_MODULE_PATH})
